### PR TITLE
Updates via typelevel-steward, use new sbt-tl feat

### PR DESCRIPTION
**Pull request by armanbilge**
1. updates now come from typelevel-steward
2. sbt-tl can generate the fmt check in CI automatically now

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@
 pull_request_rules:
 - name: merge scala-steward's PRs
   conditions:
-  - author=scala-steward
+  - author=typlevel-steward
```
Comment (joroKr21) [inline .mergify.yml:11]:
```suggestion
  - author=typelevel-steward
```

File: build.sbt [hunk 2]
```diff
@@ -80,7 +80,10 @@ ThisBuild / developers := List(
 )
 
 ThisBuild / tlCiReleaseBranches := Seq("master")
-ThisBuild / mergifyStewardConfig ~= (_.map(_.copy(mergeMinors = true)))
-ThisBuild / githubWorkflowBuild :=
-  WorkflowStep.Sbt(List("scalafmtCheckAll", "scalafmtSbtCheck"), name = Some("Check formatting")) +:
-    (ThisBuild / githubWorkflowBuild).value
+ThisBuild / mergifyStewardConfig := Some(
+  MergifyStewardConfig(
+    author = "typlevel-steward",
```
Comment (joroKr21) [inline build.sbt:85]:
```suggestion
    author = "typelevel-steward",
```

**Review by joroKr21 — CHANGES_REQUESTED** [review]

**Review by joroKr21 — APPROVED** [review]

**Comment by armanbilge** [comment]
I'm so sorry, did I literally make the same typo twice! 🤦

**Comment by joroKr21** [comment]
> I'm so sorry, did I literally make the same typo twice! 🤦

Copy pasta which usually prevents typos but in this case duplicates it 😄

**Pull request merged**